okhttp/src/main/java/okhttp3/CertificatePinner.java

```diff
@@ -183,68 +194,103 @@ public void check(String hostname, Certificate... peerCertificates)
   }
 
   /**
-   * Returns list of matching certificates' pins for the hostname or {@code null} if hostname does
-   * not have pinned certificates.
+   * Returns list of matching certificates' pins for the hostname. Returns an empty list if the
+   * hostname does not have pinned certificates.
    */
-  Set<ByteString> findMatchingPins(String hostname) {
-    Set<ByteString> directPins = hostnameToPins.get(hostname);
-    Set<ByteString> wildcardPins = null;
-
-    int indexOfFirstDot = hostname.indexOf('.');
-    int indexOfLastDot = hostname.lastIndexOf('.');
-
-    // Skip hostnames with one dot symbol for wildcard pattern search
-    //   example.com   will  be skipped
-    //   a.example.com won't be skipped
-    if (indexOfFirstDot != indexOfLastDot) {
-      // a.example.com -> search for wildcard pattern *.example.com
-      wildcardPins = hostnameToPins.get("*." + hostname.substring(indexOfFirstDot + 1));
-    }
-
-    if (directPins == null && wildcardPins == null) return null;
-
-    if (directPins != null && wildcardPins != null) {
-      Set<ByteString> pins = new LinkedHashSet<>();
-      pins.addAll(directPins);
-      pins.addAll(wildcardPins);
-      return pins;
+  List<Pin> findMatchingPins(String hostname) {
+    List<Pin> result = Collections.emptyList();
+    for (Pin pin : pins) {
+      if (pin.matches(hostname)) {
+        if (result.isEmpty()) result = new ArrayList<>();
+        result.add(pin);
+      }
     }
-
-    if (directPins != null) return directPins;
-
-    return wildcardPins;
+    return result;
   }
 
   Builder newBuilder() {
     return new Builder(this);
   }
 
-  /**
-   * Returns the SHA-1 of {@code certificate}'s public key. This uses the mechanism Moxie
-   * Marlinspike describes in <a href="https://github.com/moxie0/AndroidPinning">Android
-   * Pinning</a>.
-   */
+  /** Returns the SHA-256 of {@code certificate}'s public key. */
```

Comment:
Should we document that this along the lines of that the method returns the latest pin hash format supported by this class. Technically you're making a breaking API change here, but I think we can gloss over it with documentation to something like that effect.